### PR TITLE
Add dark-mode feature across the page

### PR DESCRIPTION
**Pull request by abdullahceylan**
# Summary

This PR adds the dark-mode feature (#212) across the page rather than only for the editor. The new dark mode control has been moved to the `Nav`.

# Screenshots
Screenshot 2021-10-27 at 23 15 13

Screenshot 2021-10-27 at 23 14 58

### CODE REVIEW

File: hooks/useDarkMode.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const isBrowser = typeof window !== 'undefined'
+
+export default function useDarkMode() {
+  // Define the localstorage key
+  const storageName = 'editor-color-theme'
+  // Get value from localstorage by key
+  const storageTheme = isBrowser ? window.localStorage.getItem(storageName) : 'light'
+
+  const [enabled, setEnabled] = useState(storageTheme !== 'light')
```
[the remark this answers is not in the document]
Comment (abdullahceylan) [reply hooks/useDarkMode.js:11]: You're right, I missed that. Updating.

File: hooks/useDarkMode.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  // Fire off effect that add/removes dark mode class
+  useEffect(() => {
+    const className = 'dark'
+    const element = window.document.documentElement
+    if (enabled) {
+      localStorage.setItem(storageName, 'vs-dark')
```
[the remark this answers is not in the document]
Comment (abdullahceylan) [reply hooks/useDarkMode.js]: Done!

File: hooks/useDarkMode.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (abdullahceylan) [reply hooks/useDarkMode.js]: Done!

File: hooks/useDarkMode.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react'
+
+const isBrowser = typeof window !== 'undefined'
+
+export default function useDarkMode() {
+  // Define the localstorage key
+  const storageName = 'editor-color-theme'
```
[the remark this answers is not in the document]
Comment (abdullahceylan) [reply hooks/useDarkMode.js]: Done!

**Comment by abdullahceylan** [comment]
- Default theme is `light` now
- The editor theme is always `dark`
- The section buttons on the left are a bit darker now
- The reset icon colour also changes depending on the theme

**Review by octokatherine — APPROVED** [review]
thanks for making those updates! @abdullahceylan

**Pull request merged**